modules/audigentRtdProvider.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * This module adds audigent provider to the real time data module
+ * The {@link module:modules/realTimeData} module is required
+ * The module will fetch segments from audigent server
+ * @module modules/audigentRtdProvider
+ * @requires module:modules/realTimeData
+ */
+
+/**
+ * @typedef {Object} ModuleParams
+ * @property {string} siteKey
+ * @property {string} pubKey
+ * @property {string} url
+ * @property {?string} keyName
+ * @property {number} auctionDelay
+ */
+
+import {config} from '../src/config.js';
+import {getGlobal} from '../src/prebidGlobal';
+import * as utils from '../src/utils';
+import {submodule} from '../src/hook';
+import {ajax} from '../src/ajax';
+
+/** @type {string} */
+const MODULE_NAME = 'realTimeData';
+
+/** @type {ModuleParams} */
+let _moduleParams = {};
+
+/**
+ * XMLHttpRequest to get data form audigent server
+ * @param {string} url server url with query params
+ */
+function getSegments(adUnits, onDone) {
+  const userIds = (getGlobal()).getUserIds();
+  let tdid = null;
+
+  if (userIds && userIds['tdid']) {
+    tdid = userIds['tdid'];
+  } else {
+    onDone({});
+  }
+
+  const url = `//seg.ad.gt/api/v1/rtb_segments?tdid=${tdid}`;
```

Comment:
This has to be hardcoded to HTTPS - protocol relative URLs are no longer allowed.